deepchecks/tabular/checks/model_evaluation/performance_bias.py

```diff
@@ -0,0 +1,549 @@
+                dataset, self.control_feature, max_segments=self.max_categories
+            )
+
+        return partitions
+
+    def _make_scores_df(self, model, dataset, scorer, partitions, model_classes):
+        """
+        Compute performance scores.
+
+        Compute performance scores disaggregated by `feature` and `control_feature` categories,
+        and averaged over `feature` for each `control_feature` level. Also computes subgroup size.
+        """
+        classwise = is_classwise(scorer, model, dataset)
+
+        scores_df = expand_grid(**partitions, _scorer=[scorer])
+
+        # Get datasets for each subgroup
+        scores_df["_dataset"] = scores_df.apply(lambda x: combine_filters(x[partitions.keys()], dataset.data), axis=1)
+
+        def score(data, model, scorer):
+            if (len(data) == 0) or (len(data) < self.min_subgroup_size):
+                if classwise:
+                    return {cls: np.nan for cls in model_classes}
+                else:
+                    return np.nan
+            return scorer(model, dataset.copy(data))
+
+        def apply_scorer(x):
+            return score(x["_dataset"], model, x["_scorer"])
+
+        # Compute subgroup scores and baseline scores
+        scores_df["_score"] = scores_df.apply(apply_scorer, axis=1)
+        if self.control_feature is not None:
+            control_scores = {
+                x.label: score(x.filter(dataset.data), model, scorer) for x in scores_df[self.control_feature].unique()
+            }
+            control_count = {x.label: len(x.filter(dataset.data)) for x in scores_df[self.control_feature].unique()}
+            scores_df["_baseline"] = scores_df.apply(lambda x: control_scores[x[self.control_feature].label], axis=1)
+            scores_df["_baseline_count"] = scores_df.apply(
+                lambda x: control_count[x[self.control_feature].label], axis=1
+            )
+        else:
+            overall_score = score(dataset.data, model, scorer)
+            overall_len = len(dataset.data)
+            scores_df["_baseline"] = scores_df.apply(lambda x: overall_score, axis=1)
+            scores_df["_baseline_count"] = scores_df.apply(lambda x: overall_len, axis=1)
+
+        # Compute subgroup size
+        scores_df["_count"] = scores_df.apply(lambda x: len(x["_dataset"]), axis=1)
+
+        # Replace functions and datasets by their name as a string
+        scores_df["_scorer"] = scores_df.apply(lambda x: x["_scorer"].name, axis=1)
+        for col_name in partitions.keys():
+            scores_df[col_name] = scores_df.apply(lambda x, col_name=col_name: x[col_name].label, axis=1)
+
+        # Drop subgroup dataset
+        scores_df.drop(labels=["_dataset"], axis=1, inplace=True)
+
+        # For class-wise prediction, explode the scores to a row per class
+        if classwise:
+            scores_df.insert(len(scores_df.columns) - 3, "_class", scores_df.apply(lambda x: list(x["_score"]), axis=1))
```

Comment:
Why -3?

Reply:
Insert at a specific location, why does it matter?